src/external/mongo.ts

```diff
@@ -0,0 +1,62 @@
+import * as mongodb from 'mongodb';
+
+import config from '../config';
+import Logger from '../logger';
+const logger = Logger('Mongo');
+
+const CREDENTIALS = config.mongo.user && config.mongo.password ? `${config.mongo.user}:${config.mongo.password}@` : '';
+const CONNECTION = `mongodb://${CREDENTIALS}${config.mongo.host}/${config.mongo.db}`;
+const DB = config.mongo.db;
+
+const client = new mongodb.MongoClient(CONNECTION, { maxPoolSize: 100 });
+
+let connected = false;
+
+export const collections: {
+  source: {
+    collections?: mongodb.Collection;
+    datasources?: mongodb.Collection;
+    files?: mongodb.Collection;
+    hosts?: mongodb.Collection;
+    samples?: mongodb.Collection;
+    sequences?: mongodb.Collection;
+  };
+  sequenceCentric?: mongodb.Collection;
+  runs?: mongodb.Collection;
+} = { source: {} };
+
+export const getDb = async () => {
+  if (!connected) {
+    throw new Error('DB Not Connected! Connect to mongo before requesting the database.');
+  }
+  return client.db(DB);
+};
+
+export const connect = async () => {
+  if (!connected) {
```

Comment:
Not a request for change but stating my personal style preference: "never ask negatives"...
it makes for lesser cognitive burden, especially for those of us with cognitive impairments. 

e.g. I'd rather wrap the rest of this function in a (positive) `if`, and have these three lines after the if (which should return and skip them.)

Reply:
Interesting point, I hadn't thought of the negative conditional in this way, I do use them semi-regularly.

In general, I dislike wrapping long code blocks in conditional statements, it can lead to a lot of nesting that adds a lot of burden to remembering what code block you're in and what it means.

In this specific case I don't think I can flip the section wrapped in the conditional... I'm ensuring this code block only runs once while the rest should run every time, after the connection occurs.